internal/api/client.go

```diff
@@ -150,38 +151,47 @@ func (c *Client) ExecuteWithRetry(
 		switch apiRes.StatusCode {
 		case endpointCfg.SuccessStatus:
 			// success case
+		case http.StatusTooManyRequests:
+			header := apiRes.Header.Get("Retry-After")
+			retryAfter, err := strconv.Atoi(header)
+			if err != nil {
+				return nil, dur, fmt.Errorf("error parsing Retry-After value from response header")
+			}
+			dur = time.Second * time.Duration(retryAfter)
```

Comment:
stagger the retries with a random buffer added to this duration. not too big, ms worth...
retry limit: 100/min
API limit is per tenant

one terraform provider running, but what if the customer is creating 100 users at a time...? or such requests.